### DISCUSSION TITLE
Plex - Dell Tiny PCs - Intel Coffeelake GPU - card0/card1

### DISCUSSION THREAD

**Discussion by vikramwalia** [tteck/Proxmox#3096] [Help Needed]
I used ttech's scripts to install plex on one of my proxmox machine( Thank you for providing that , it is amazing) . That script also creates card1/0 and passes it through to the plex lxc. In my not so smart brain.. I used that script to create plex lxcs on my other proxmox servers as well and then delete them all except one. The problem now is that 2 proxmox servers have it as card0 and other 2 have it as card1.

pmox04 - card1
pmox05 - card1
pmox06 - card0
pmox07 - card0

Is there was way I can rename the card0 to card1 ? Plex is the only lxc using it and I would love to have all machines on card1 if possible so that when I migrate or use HA, it would move over without any issues. Can I create a symlink or point card1 to card0 ? any ideas suggestions as welcome.

**Comment by tteck** [comment]
The script checks for `card0` if not found sets it to `card1`
https://github.com/tteck/Proxmox/blob/5e515c1e23156617e16219668d5dea8e839542ce/misc/build.func#L607-L621

With a recent update, it changed some systems from `card0` to `card1` for reasons I don't know. This is why I had to add a check.

**Reply by tteck** [reply]
BTW, my system was also affected by the card rename 🤷🏻‍♂️

**Comment by vikramwalia** [comment]
Thanks much for the reply, is there a way to do a symlink or something similar ? my Goal is to get HA working.

**Reply by tteck** [reply]
Good question, I've never tried a symbolic link for a device before.

**Reply by tteck** [reply]
One could test this by `ln -s /dev/dri/card1 /dev/dri/card0
` This command creates a symbolic link named `card0` in the `/dev/dri` directory, pointing to `/dev/dri/card1`.

EDIT: fixed command (`/dev/dri`)

**Reply by vikramwalia** [reply]
I was able to create the symlink and the LXC migrated over without an issues. yay ! However the symlink does not survive a reboot. I will put that in script to be done manually but thank you that was an amazing easy solution !!

```
root@pmox07:/dev/dri# ls -ltr
total 0
crw-rw---- 1 root render 226, 128 May 29 19:19 renderD128
crwxrwxrwx 1 root video  226,   0 May 29 19:19 card0
drwxr-xr-x 2 root root         80 May 29 19:19 by-path
lrwxrwxrwx 1 root root         14 May 29 19:22 card1 -> /dev/dri/card0
```

**Reply by tteck** [reply]
systemd service?

**Reply by tteck** [reply]
This _may_ work.

```
nano /etc/systemd/system/create-symlink.service
```
```
[Unit]
Description=Create symbolic link for graphics card

[Service]
Type=oneshot
ExecStart=/bin/ln -s /dev/dri/card1 /dev/dri/card0

[Install]
WantedBy=multi-user.target
```
```
systemctl enable --now create-symlink.service
```

**Reply by vikramwalia** [reply]
That would be the easiest. I am going to test out systemd and also look at udev as well.

**Reply by pedro1977** [reply]
Any update on this? I did a test migration and encountered the same issue. Is a symlink the best set and forget solution?

**Reply by vikramwalia** [reply]
Yes, symlink is a set it forget it solution. At least for me it survived two reboots and it works.

**Reply by psilantropy** [reply]
Thanks tteck. 3 NUC11 nodes, one stuck on card1, others card0.
Symlink works perfectly.

**Comment by vikramwalia** [comment]
On another note, I did ask exactly the same question in proxmox forums as well. The suggested path was to look at udev rules.  I am going to look into that as well to see if we can have a more permanent solution i.e. pin card0 vs having card1.

Link : https://forum.proxmox.com/threads/plex-dell-tiny-pcs-intel-coffeelake-gpu-card0-card1.148146/#post-669850

**Comment by avluis** [comment]
This issue is the result of early initialization of several framebuffer drivers (load order something along those lines, prior refs are not turning up much) but essentially as was discovered here, setup a symlink from card1 pointing to card0 and that will address this (that is until another kernel update that might address this).

This affected my N5xxx, N6xxx nodes and a Lenovo Tiny with a Gen 9 i5 (basically anything with a GPU, built-in or addon).
Other refs on this: https://bbs.archlinux.org/viewtopic.php?id=287936 | https://www.linux.org/threads/video-card-being-being-added-as-card1-instead-of-card0.46280/ | https://bbs.archlinux.org/viewtopic.php?id=288578
